client/app/components/modal/layout/ModalBody.jsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import {Modal} from "react-bootstrap";
+import _ from "lodash";
+import FormTextField from "./body/FormTextField";
+import FormDateField from "./body/FormDateField";
+import FormLocationField from "./body/FormLocationField";
+import ErrorField from "./ErrorField";
+
+const { func, array, bool } = React.PropTypes;
+
+export default class ModalBody extends React.Component {
+  static propTypes = {
+    updateParam:      func.isRequired,
+    conferenceRooms:  array.isRequired,
+    showErrorMessage: bool
+  };
+
+  static defaultProps = {
+    showErrorMessage: false
+  };
+
+  constructor(props) {
+    super(props);
+
+    _.bindAll(this,
+      ['handleTextFieldChange', 'handleLocationChange', 'handleStartTimeChange', 'handleEndTimeChange']);
+  }
+
+  handleTextFieldChange(e) {
+    const value = e.target.value;
+    const name = e.target.name;
+
+    this.props.updateParam(name, value);
+  }
+
+  handleLocationChange(e) {
+    this.props.updateParam('conferenceRoomId', e.target.value);
+  }
+
+  handleStartTimeChange(e) {
+    if (e !== 'Invalid date') {
+      this.props.updateParam('startTime', e);
+    }
+  }
+
+  handleEndTimeChange(e) {
```

Comment:
To be honest this func is exactly the same as `handleStartTimeChange` apart from `updateParam` first arg. How about we'll extract the body to separate func and call it with proper prop name.